File: Android project/ViewPager/app/src/main/java/com/mirkamol/viewpager/adapters/ViewPagerAdapter.java
package com.mirkamol.viewpager.adapters;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.ArrayList;
import java.util.List;

public class ViewPagerAdapter extends FragmentPagerAdapter {

    private final List<Fragment> fragments = new ArrayList<>();
    private final List<String> fragmentTitle = new ArrayList<>();

    public ViewPagerAdapter(@NonNull FragmentManager fragmentManager){
        super(fragmentManager);
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    public void add(Fragment fragment, String title){
        fragments.add(fragment);
        fragmentTitle.add(title);
    }

    public int getCount(){
        return fragments.size();
    }

    public CharSequence getPageTitle(int position){
        return fragmentTitle.get(position);
    }
}

File: Android project/FragmentTask45/app/src/main/java/com/mirkamol/fragmenttask4_5/adapter/ViewPagerAdapter.kt
package com.example.fragmenttask5.adapter

import androidx.fragment.app.*

class ViewPagerAdapter(var fragment: ArrayList<Fragment>, var fragmentManager: FragmentManager, var fragmentTitle : ArrayList<String>):
    FragmentStatePagerAdapter(fragmentManager, FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT) {
    override fun getCount(): Int {
        return fragment.size
    }

    override fun getItem(position: Int): Fragment {
        return fragment[position]
    }

    override fun getPageTitle(position: Int): CharSequence {
        return fragmentTitle[position]
    }

}